src/common/AbstractObject.h

```diff
@@ -301,7 +301,7 @@ namespace COMMON_NS
 
 		enum hdfDatatypeEnum { UNKNOWN = 0, DOUBLE = 1, FLOAT = 2, LONG_64 = 3, ULONG_64 = 4, INT = 5, UINT = 6, SHORT = 7, USHORT = 8, CHAR = 9, UCHAR = 10};
 
-		virtual ~AbstractObject() = default;
+		DLL_IMPORT_OR_EXPORT virtual ~AbstractObject() = default;
```

Comment:
I don't understand why we should export that.
Since we don't inherit from AbstractObject in Java, niether C#, I am wondering what's the need for this export.

Reply:
AbstractHdfProxy may be inherited in target languages (this is the main purpose of this pull request). In such case, overriding AbstractObject::getXmlNamespace() is required. Thus, AbstractObject needs to be a director base class. To not export the virtual destructor in a director base class leads to a warning (205) during the Swig processing.

Reply:
Thanks. That's now perfectly clear.
Would it be possible please to copy paste this reason in the the swigModule.i just before to declare AbstractObject as a director.
This is the kind of reason which I am about to forget in some months/years unfortunately.

Reply:
Sure. Done.